### PR TITLE
feat(uptime): add trace_sampled field to uptime kafka message

### PR DESCRIPTION
**Pull request by JoshFerge**
part of https://github.com/getsentry/team-uptime/issues/22

depends on package with https://github.com/getsentry/sentry-kafka-schemas/pull/351 going out.

### CODE REVIEW

File: src/sentry/uptime/subscriptions/tasks.py [hunk 1]
```diff
@@ -95,6 +95,7 @@ def uptime_subscription_to_check_config(
         "timeout_ms": subscription.timeout_ms,
         "request_method": subscription.method,  # type: ignore[typeddict-item]
         "request_headers": headers,
+        "trace_sampling": subscription.trace_sampling,
```
Comment (evanpurkhiser) [inline src/sentry/uptime/subscriptions/tasks.py:98]: You'll have to release and bump the `sentry-kafka-schemas` package to not get mypy errors here I think.

Comment (JoshFerge) [reply src/sentry/uptime/subscriptions/tasks.py:98]: it's funny i actually didn't get any mypy errors, but yeah, i called this out in the the PR description.

Comment (evanpurkhiser) [reply src/sentry/uptime/subscriptions/tasks.py:98]: hmmm, why doesn't this produce mypy errors?

Comment (JoshFerge) [reply src/sentry/uptime/subscriptions/tasks.py:98]: because of the other ignores added above. i think it makes the ignores apply to the whole statement

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**